src/test/java/io/appium/java_client/pagefactory_tests/TimeOutTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.pagefactory_tests;
+
+import static io.appium.java_client.ChromeDriverPathUtil.getChromeDriver;
+import static io.appium.java_client.pagefactory.AppiumFieldDecorator.DEFAULT_TIMEOUT;
+import static io.appium.java_client.pagefactory.AppiumFieldDecorator.DEFAULT_TIMEUNIT;
+import static java.lang.Math.abs;
+import static java.lang.String.format;
+import static java.lang.System.setProperty;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.junit.Assert.assertThat;
+import static org.openqa.selenium.support.PageFactory.initElements;
+
+import io.appium.java_client.pagefactory.AppiumFieldDecorator;
+import io.appium.java_client.pagefactory.TimeOutDuration;
+import io.appium.java_client.pagefactory.WithTimeout;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeDriverService;
+import org.openqa.selenium.support.FindAll;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.Calendar;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class TimeOutTest {
+
+    private static final long ACCEPTABLE_DELTA_MILLS = 1500;
+    private static final String MESSAGE = "Check difference from the expected waiting duration %s %s";
+
+    private WebDriver driver;
+
+    @FindAll({
+            @FindBy(className = "ClassWhichDoesNotExist"),
+            @FindBy(className = "OneAnotherClassWhichDoesNotExist")})
+    private List<WebElement> stubElements;
+
+    @WithTimeout(time = 5, unit = SECONDS)
+    @FindAll({@FindBy(className = "ClassWhichDoesNotExist"),
+            @FindBy(className = "OneAnotherClassWhichDoesNotExist")})
+    private List<WebElement> stubElements2;
+
+    private TimeOutDuration timeOutDuration;
+
+    private static long getBenchMark(List<WebElement> stubElements) {
```

Comment:
One can make this method more generic by providing a lambda function to benchmark:

```java
private static long getBenchmark(Runnable dstFunc) {
final long startTime = System.nanoTime();
dstFunc.run();
return System.nanoTime() - startTime;
}
```